controller/volume_controller.go

```diff
@@ -1567,10 +1947,104 @@ func (vc *VolumeController) getReplenishReplicasCount(v *longhorn.Volume, rs map
 		}
 	}
 
-	if v.Spec.NumberOfReplicas < usableCount {
-		return 0
+	switch {
+	case v.Spec.NumberOfReplicas < usableCount:
+		return 0, ""
+	case v.Spec.NumberOfReplicas > usableCount:
+		return v.Spec.NumberOfReplicas - usableCount, ""
+	case v.Spec.NumberOfReplicas == usableCount:
+		adjustNodeAffinity := ""
+		adjustCount := vc.getReplicaCountForAutoBalanceLeastEffort(v, e, rs, vc.getReplicaCountForAutoBalanceZone)
+		if adjustCount == 0 {
+			adjustCount = vc.getReplicaCountForAutoBalanceLeastEffort(v, e, rs, vc.getReplicaCountForAutoBalanceNode)
+		}
+		var nCandidates []string
```

Comment:
If the adjust count is not zero, there is no need to continue. Hence we can directly return. e.g.,:
```suggest
if adjustCount := vc.getReplicaCountForAutoBalanceLeastEffort(v, e, rs, vc.getReplicaCountForAutoBalanceZone); adjustCount != 0 {
    return adjustCount, ""
}
if adjustCount := vc.getReplicaCountForAutoBalanceLeastEffort(v, e, rs, vc.getReplicaCountForAutoBalanceNode); adjustCount != 0 {
    return adjustCount, ""
}
adjustNodeAffinity := ""
```

Reply:
Thanks. Fixed.